### ISSUE TITLE
@node-red/editor-client	Css overrides global styles for HTML elements

### ISSUE THREAD

**Issue by KamalAman** [node-red/node-red#2110]
### What are the steps to reproduce?
@node-red/editor-client defines global css styles that effect all DOM elements

### What happens?
@node-red/editor-client defines global overrides for css styles because it uses styles extracted from Bootstrap v2.3.2.

For example all textareas get the following style applied to them:
```css
textarea,input[type="text"],input[type="password"],input[type="datetime"],input[type="datetime-local"],input[type="date"],input[type="month"],input[type="time"],input[type="week"],input[type="number"],input[type="email"],input[type="url"],input[type="search"],input[type="tel"],input[type="color"],div[contenteditable="true"],.uneditable-input {
    background-color: #ffffff;
    border: 1px solid #ccc;
    -webkit-box-shadow: inset 0 1px 1px rgba(0,0,0,0.075);
    -moz-box-shadow: inset 0 1px 1px rgba(0,0,0,0.075);
    box-shadow: inset 0 1px 1px rgba(0,0,0,0.075)
}
```

This results in unwanted styling being applied. For example if you wanted to use a UI component libraries such as material-ui or define your own custom component, you need to apply ani-styles to counteract these styles.

### What do you expect to happen?
No global element styles should be defined for HTML elements. Specify at least 1 class name or class name descendant for styles e.g.  `.form-row > textarea` 

The following is the list of styles that should be made more restrictive 
```css
button,input,select,div[contenteditable="true"],textarea
button,div[contenteditable="true"],input
button::-moz-focus-inner,div[contenteditable="true"]::-moz-focus-inner,input::-moz-focus-inner
button,html input[type="button"],input[type="reset"],input[type="submit"]
label,select,button,input[type="button"],input[type="reset"],input[type="submit"],input[type="radio"],input[type="checkbox"]
input[type="search"]
input[type="search"]::-webkit-search-decoration,input[type="search"]::-webkit-search-cancel-button
textarea
form
fieldset
legend
legend small
label,input,div[contenteditable="true"],button,select,textarea
input,div[contenteditable="true"],button,select,textarea
label 
select,textarea,input[type="text"],input[type="password"],input[type="datetime"],input[type="datetime-local"],input[type="date"],input[type="month"],input[type="time"],input[type="week"],input[type="number"],input[type="email"],input[type="url"],input[type="search"],input[type="tel"],input[type="color"],div[contenteditable="true"],.uneditable-input
input,textarea,div[contenteditable="true"],.uneditable-input
textarea 
textarea,input[type="text"],input[type="password"],input[type="datetime"],input[type="datetime-local"],input[type="date"],input[type="month"],input[type="time"],input[type="week"],input[type="number"],input[type="email"],input[type="url"],input[type="search"],input[type="tel"],input[type="color"],div[contenteditable="true"],.uneditable-input 
textarea:focus,input[type="text"]:focus,input[type="password"]:focus,input[type="datetime"]:focus,input[type="datetime-local"]:focus,input[type="date"]:focus,input[type="month"]:focus,input[type="time"]:focus,input[type="week"]:focus,input[type="number"]:focus,input[type="email"]:focus,input[type="url"]:focus,input[type="search"]:focus,input[type="tel"]:focus,input[type="color"]:focus,div[contenteditable="true"]:focus,.uneditable-input:focus
input[type="radio"],input[type="checkbox"]
input[type="file"],input[type="image"],input[type="submit"],input[type="reset"],input[type="button"],input[type="radio"],input[type="checkbox"] 
select,input[type="file"] 
select 
select[multiple],select[size] 
select:focus,input[type="file"]:focus,input[type="radio"]:focus,input[type="checkbox"]:focus
```

### Please tell us about your environment:

- [ ] Node-RED version: 0.20.3
- [ ] node.js version: 10.10
- [ ] npm version: 6.4.1
- [ ] Platform/OS: macOX
- [ ] Browser: Chrome

**Comment by knolleary** [comment]
We provide a default set of styles, derived from bootstrap, for all elements. They ought to be a starting point for your own elements. Using something like material-ui would create node edit dialogs that are completely inconsistent with the rest of the UI and other nodes.

That said, I'm well aware we let bootstrap do a lot of the heavy lifting and we really should have our own, more clearly defined, style guide for nodes.

Any changes we make to these default styles has to be done very carefully to not break the appearance of all existing nodes that rely on them.

There is also longer-term task to tidy up all the styling so that it only applies to elements within the node-red created UI. This would allow the editor to be embedded as a component into an existing page.

**Comment by KamalAman** [comment]
Embedding the editor as a component into an existing page is precisely what we are trying to do, however as soon as we include the style sheet it interferes with other ui elements. 

[Scoped CSS](https://css-tricks.com/saving-the-day-with-scoped-css/) might make this possible

**Comment by knolleary** [comment]
Fixed by #2158 in the dev branch for the 1.0 release. This is a massive set of changes to the css. We will have a migration guide nearer to the 1.0 release.

As it relates to this issue, the main change is all of the base styles of default elements now only applies to elements within a defined set of classes/ids - `#red-ui-editor` being the main container of the editor.

The way the editor builds on the page has also changed; the template page has been drastically simplified - and the code injects the editor into the div with the `#red-ui-editor` id. This allows custom templates to be used that places that div wherever it is needed on the page.

This does *not* allow for multiple instances of the editor on the page.

And of course, there may yet be bugs as it isn't a primary use case we test. But a huge step forwards.

We'll have a beta of 1.0 in the next week to start getting feedback from users.
